Build fails using boost 1.69.0-1

Build failed using latest version of boost 1.69.0-1 and boost-libs 1.69.0-1.



`[ 58%] Built target FairingAlgorithms
/builds/libpsalm/mesh.cpp: In member function 'bool psalm::mesh::load_obj(std::istream&)':
/builds/libpsalm/mesh.cpp:640:27: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
       if((V.size()+index) >= 0)
          ~~~~~~~~~~~~~~~~~^~~~
In file included from /builds/libpsalm/SubdivisionAlgorithms/Loop.cpp:8:
/builds/libpsalm/SubdivisionAlgorithms/Loop.h: In member function 'virtual bool psalm::Loop::set_weights(psalm::SubdivisionAlgorithm::weights)':
/builds/libpsalm/SubdivisionAlgorithms/Loop.h:29:28: warning: unused parameter 'new_weights' [-Wunused-parameter]
   bool set_weights(weights new_weights)
                    ~~~~~~~~^~~~~~~~~~~
In file included from /usr/include/boost/random/detail/integer_log2.hpp:19,
                 from /usr/include/boost/random/detail/large_arithmetic.hpp:19,
                 from /usr/include/boost/random/detail/const_mod.hpp:23,
                 from /usr/include/boost/random/detail/seed_impl.hpp:26,
                 from /usr/include/boost/random/mersenne_twister.hpp:30,
                 from /usr/include/boost/uuid/random_generator.hpp:17,
                 from /usr/include/boost/uuid/uuid_generators.hpp:17,
                 from /builds/libpsalm/libpsalm.cpp:13:
/usr/include/boost/pending/integer_log2.hpp:7:59: note: #pragma message: This header is deprecated. Use <boost/integer/integer_log2.hpp> instead.
 BOOST_HEADER_DEPRECATED("<boost/integer/integer_log2.hpp>");
                                                           ^
In file included from /builds/libpsalm/libpsalm.cpp:18:
/builds/libpsalm/SubdivisionAlgorithms/Liepa.h: In member function 'virtual bool psalm::Liepa::set_weights(psalm::SubdivisionAlgorithm::weights)':
/builds/libpsalm/SubdivisionAlgorithms/Liepa.h:31:28: warning: unused parameter 'new_weights' [-Wunused-parameter]
   bool set_weights(weights new_weights)
                    ~~~~~~~~^~~~~~~~~~~
make[1]: *** [CMakeFiles/Makefile2:254: SegmentationAlgorithms/CMakeFiles/SegmentationAlgorithms.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:144: CMakeFiles/psalm.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:364: TriangulationAlgorithms/CMakeFiles/TriangulationAlgorithms.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:309: SubdivisionAlgorithms/CMakeFiles/SubdivisionAlgorithms.dir/all] Error 2
make: *** [Makefile:84: all] Error 2`

--- a/edge.cpp
+++ b/edge.cpp
@@ -204,7 +204,7 @@ bool edge::is_on_boundary()
 	if(boost::logic::indeterminate(boundary))
 		boundary = (f == NULL) || (g == NULL);
 
-	return(boundary == true);
+	return(bool(boundary) == true);
 }
 
 /*!

--- a/face.cpp
+++ b/face.cpp
@@ -209,7 +209,7 @@ bool face::is_obtuse()
 		obtuse = (a*a+b*b < c*c) || (b*b* + c*c < a*a) || (c*c + a*a < b*b);
 	}
 
-	return(obtuse == true);
+	return(bool(obtuse) == true);
 }
 
 } // end of namespace "psalm"

